packages/suite/src/hooks/wallet/coinmarket/form/useCoinmarketExchangeForm.ts

```diff
@@ -657,17 +657,21 @@ export const useCoinmarketExchangeForm = ({
         };
     }, []);
 
-    const filteredQuotes = useMemo(
-        () => getQuotesByRateType(rateType, innerQuotes, exchangeInfo),
-        [innerQuotes, rateType, exchangeInfo],
+    const filteredCexQuotes = useMemo(
+        () => getCexQuotesByRateType(rateType, innerQuotes, exchangeInfo),
+        [rateType, innerQuotes, exchangeInfo],
     );
 
-    // set new default selected quote if the current one is not in the filtered quotes
+    const dexQuotes = useMemo(() => innerQuotes?.filter(q => q.isDex), [innerQuotes]);
+
+    // handle edge case when there are no longer quotes of selected exchange type
     useEffect(() => {
-        if (!filteredQuotes?.length) setSelectedExchangeQuote(undefined);
-        else if (!filteredQuotes.some(quote => quote.orderId === selectedQuote?.orderId))
-            setSelectedExchangeQuote(filteredQuotes[0]);
-    }, [filteredQuotes, selectedQuote?.orderId, setSelectedExchangeQuote]);
+        if (exchangeType === 'DEX' && !dexQuotes?.length && filteredCexQuotes?.length) {
+            setValue('exchangeType', 'CEX');
+        } else if (exchangeType === 'CEX' && !filteredCexQuotes?.length && dexQuotes?.length) {
+            setValue('exchangeType', 'DEX');
+        }
+    }, [dexQuotes, exchangeType, filteredCexQuotes, setValue]);
```

Comment:
I am not sure about that.
1. Move the code block to the same context. `useEffect` with `useEffects` etc.
2. Do we need new parameters such as `dexQuotes` and `filteredCexQuotes`?
- what about creating a function with input parameters `quotes`, `rateType` & `exchangeType`, and return quotes that will be filtered quotes according to `rateType` and `exchangeType`? An example can be sorting according to `paymentMethod` in buy.
```
 const filterExchangeQuotes = (quotes, rateType, exchangeType) => {
   if (exchangeType === 'DEX') return quotes // logic for floating
   if (rateType === 'floating') return quotes // logic for floating
   
   return quotes // logic for fixed
 }
```

Reply:
I need those two different collections (dex and filteredCex) on form screen. I think concatenating them together would increase complexity in the view.